packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -95,6 +95,7 @@ abstract class Pub {
     bool generateSyntheticPackage = false,
     String flutterRootOverride,
     bool checkUpToDate = false,
+    bool shouldSkipThirdPartyGenerator = true,
```

Comment:
nit: I would add a quick blurb on what `shouldSkipThirdPartyGenerator` does, like:

`if [shouldSkipThirdPartyGenerator] is true, the package config will be read if it exists and the overall pub get skipped if the generator field is a value other than "pub".`